assets/js/theme/global/currency-selector.js

```diff
@@ -1,5 +1,57 @@
-export default function () {
+import swal from './sweet-alert';
+import utils from '@bigcommerce/stencil-utils';
+
+export default function (cartId) {
+    function changeCurrency(url, currencyCode) {
+        $.ajax({
+            url,
+            method: 'POST',
+            data: JSON.stringify({ currencyCode }),
+        }).then(() => {
+            window.location.reload();
+        });
+    }
+
     $(document.body).on('click', '.currencySelector', () => {
         $('.currency-selection-list').toggleClass('active');
     });
+
+    $(document.body).on('click', '.dropdown-menu-item a', event => {
+        if (!cartId) {
+            return;
+        }
+
+        event.preventDefault();
+
+        utils.api.cart.getCart({ cartId }, (err, response) => {
+            let showWarning = false;
+            response.discounts.forEach((discount) => {
```

Comment:
You can write `response.discounts.some(discount => discount > 0)`. It exits the loop as soon as there's a match. With that, you can combine all checks into one so you don't need to do other checks if one of those conditions is true. i.e.:

```js
const showWarning = response.discounts.some(discount => discount > 0) ||
  response.coupons.length > 0 ||
  response.lineItems.giftCertificates.length > 0;
```